### PR TITLE
Support for multiple custom rules directories in ESLint checker

### LINKED ISSUE #1016: Flycheck's ESLint checker should support multiple custom rules directories
Flycheck currently offers `flycheck-eslint-rulesdir` to specify a directory containing custom rules.

However, ESLint supports multiple rules directories, [by passing the `--rulesdir` option multiple times](http://eslint.org/docs/user-guide/command-line-interface#specifying-rules-and-plugins).

To support this in the Flycheck checker, I would suggest creating an alias called `flycheck-eslint-rulesdirs` that is a customizable list, deprecating `flycheck-eslint-rulesdir`, and handling the case where `flycheck-eslint-rulesdirs` contains a string when using it to construct the ESLint command.

That should provide the desired semantics, be backwards-compatible, and encourage users to update their configurations appropriately.

Does this sound reasonable?

### PR DESCRIPTION
**Pull request by Simplify**
Fixes #1016.

### CODE REVIEW

**Comment by lunaryorn** [comment]
@Simplify Thanks. But shouldn't we call it -rules-directories now?

**Comment by Simplify** [comment]
@lunaryorn done.

**Comment by lunaryorn** [comment]
LGTM. @cpitclaudel?

File: flycheck.el [hunk 1]
```diff
@@ -7257,20 +7257,18 @@ See URL `http://www.jshint.com'."
...
 
 Refer to the ESLint manual at URL
 `https://github.com/eslint/eslint/tree/master/docs/command-line-interface#--rulesdir'
-for more information about the custom directory."
-  :type '(choice (const :tag "No custom rules directory" nil)
-                 (directory :tag "Custom rules directory"))
-  :safe #'stringp
-  :package-version '(flycheck . "0.16"))
+for more information about the custom directories."
+  :type '(repeat (directory :tag "Custom rules directories"))
```
Comment (cpitclaudel) [inline flycheck.el]: Should this be "Custom rules directory" instead of directories? There's a single directory per entry in the list, right?

**Comment by cpitclaudel** [comment]
One small comment; other than that, LGTM.

**Comment by lunaryorn** [comment]
@Simplify Please address @cpitclaudel's comment, and then feel free to merge :blush:

**Comment by Simplify** [comment]
@cpitclaudel Done.

@lunaryorn:

travis-ci EMACS_VERSION=snapshot failed:

```
3.5 is not installed; attempting download
bzip2: (stdin) is not a bzip2 file.
tar: Child returned status 2
tar: Error is not recoverable: exiting now
The command "sudo tar xjf python-3.5.tar.bz2 --directory /" failed and exited with 2 during .
```

**Comment by cpitclaudel** [comment]
I think it might have been a fluke. I restarted the build.

**Comment by cpitclaudel** [comment]
Yup, all safe :)

**Comment by lunaryorn** [comment]
@cpitclaudel @Simplify Thanks 👍

**Pull request merged**